sources/editor/Stride.GameStudio/Helpers/ApplicationInitializer.cs

```diff
@@ -0,0 +1,374 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Security;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Threading;
+using Stride.Assets.Presentation;
+using Stride.Core.Assets;
+using Stride.Core.Assets.Editor;
+using Stride.Core.Assets.Editor.Components.TemplateDescriptions.ViewModels;
+using Stride.Core.Assets.Editor.Components.TemplateDescriptions.Views;
+using Stride.Core.Assets.Editor.Services;
+using Stride.Core.Assets.Editor.Settings;
+using Stride.Core.Assets.Editor.ViewModel;
+using Stride.Core.Diagnostics;
+using Stride.Core.Extensions;
+using Stride.Core.IO;
+using Stride.Core.MostRecentlyUsedFiles;
+using Stride.Core.Presentation.Interop;
+using Stride.Core.Presentation.View;
+using Stride.Core.Presentation.ViewModel;
+using Stride.Core.Presentation.Windows;
+using Stride.Editor.Build;
+using Stride.Editor.Engine;
+using Stride.Editor.Preview;
+using Stride.GameStudio.Plugin;
+using Stride.GameStudio.Services;
+using Stride.GameStudio.View;
+using Stride.GameStudio.ViewModels;
+using Stride.Graphics;
+using Stride.Metrics;
+using Stride.PrivacyPolicy;
+using EditorSettings = Stride.Core.Assets.Editor.Settings.EditorSettings;
+using MessageBox = System.Windows.MessageBox;
+using MessageBoxButton = System.Windows.MessageBoxButton;
+using MessageBoxImage = System.Windows.MessageBoxImage;
+// Copyright (c) .NET Foundation and Contributors (https://dotnetfoundation.org/ & https://stride3d.net) and Silicon Studio Corp. (https://www.siliconstudio.co.jp)
+// Distributed under the MIT license. See the LICENSE.md file in the project root for more information.
+namespace Stride.GameStudio.Helpers
+{
+    internal static class ApplicationInitializer
```

Comment:
So, what's your reasoning here wrt extracting those functions out of program.cs ?

Reply:
My intention was to indicate that these methods are needed to initialize the app, so I created a separate class with Init() method creating facade behind all of this.

Reply:
Right, but you've just moved the entirety of the logic within `Program` to a different class. In other words, you renamed `Program` into `ApplicationInitializer` and made a new file called `Program` just for the `Main()` to call into it.

It'll be harder to follow the git history for that file, and doesn't seem like it provides a lot in exchange ?

I welcome new methods and such to clarify what's actually going on here, but moving everything to another file is hard to justify for me. If other contributors think this is a good idea, feel free to chime in.

Reply:
@Eideren Yeah, I guess you're rigth about that. I've reverted this change 👍